ReactAndroid/src/main/java/com/facebook/react/modules/datepicker/DatePickerMode.java

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+package com.facebook.react.modules.datepicker;
+
+/**
+ * Date picker modes
+ */
+public enum DatePickerMode {
+  CALENDAR("calendar"),
+  SPINNER("spinner"),
+  DEFAULT("default");
+
+  private final String mValue;
+
+  DatePickerMode(String value) {
+    mValue = value;
+  }
+
+  public static DatePickerMode fromString(String value) {
```

Comment:
ValueOf has been implemented. For that, we need to match the user values with enum constants.